compiler/dcalc/from_scopelang.ml

```diff
@@ -926,12 +970,26 @@ let translate_scope_decl
     | None -> AbortOnRound
   in
   let ctx = { ctx with date_rounding } in
-  let scope_input_var = scope_sig.scope_sig_input_var in
+  let scope_input_var =
+    Var.make (Mark.remove (ScopeName.get_info scope_name) ^ "_in")
+  in
   let scope_input_struct_name = scope_sig.scope_sig_input_struct in
   let scope_return_struct_name = scope_sig.scope_sig_output_struct in
   let pos_sigma = Mark.get sigma_info in
+  let scope_mark =
+    (* Find a witness of a mark in the definitions *)
+    match sigma.scope_decl_rules with
+    | [] ->
+      (* Todo: are we sure this can't happen in normal code ? E.g. is calling a
```

Comment:
It's a bit more involved because of the GADT for annotations, we would need to take a type witness as parameter of the function. I'll see what I can do ^^